ecosystem/typescript/sdk/src/transaction_builder/builder.ts

```diff
@@ -292,3 +295,113 @@ export class TransactionBuilderABI {
     throw new Error("Invalid ABI.");
   }
 }
+
+export type RemoteABIBuilderConfig = Partial<Omit<ABIBuilderConfig, "sender">> & {
+  sender: MaybeHexString | AccountAddress;
+};
+
+interface AptosClientInterface {
+  getAccountModules: (arg0: string) => Promise<Gen.MoveModuleBytecode[]>;
+  getAccount: (accountAddress: MaybeHexString) => Promise<Gen.AccountData>;
+  getChainId: () => Promise<number>;
+}
+
+/**
+ * This transaction builder downloads JSON ABIs from the fullnodes.
+ * It then translates the JSON ABIs to the format that is accepted by TransactionBuilderABI
+ */
+export class TransactionBuilderRemoteABI {
+  // We don't want the builder to depend on the actual AptosClient. There might be circular dependencies.
+  constructor(
+    private readonly aptosClient: AptosClientInterface,
+    private readonly builderConfig: RemoteABIBuilderConfig,
+  ) {}
```

Comment:
We already have a `TransactionBuilderABI` that looks very similar.
What if we make `fetchABI` into a standalone function so we can reuse the code through composition?

```typescript
const abis = await fetchABI(aptosClient, '0x1')
const builder = new TransactionBuilderABI(abis, config);
const rawTxn = builder.buildTransactionPayload(
  '0x1::coin::transfer,
  ['0x1::aptos_coin::AptosCoin'],
  [recipient, amount],
)

```

This way the user has the freedom to manage the lifetime of fetched abis.

Reply:
Like the idea. But for consistency's sake, these transaction builders belong to the same family. lol. I will keep it as a builder class for now.